cpp/src/slice2java/Gen.cpp

```diff
@@ -322,58 +320,37 @@ Slice::JavaVisitor::writeMarshalUnmarshalCode(
                 }
             }
 
-            TypePtr keyType = dict->keyType();
-            TypePtr valueType = dict->valueType();
+            const TypePtr keyType = dict->keyType();
+            const TypePtr valueType = dict->valueType();
             if (marshal)
             {
                 if (optionalParam)
                 {
-                    out << nl;
+                    out << nl << "if(";
```

Comment:
We likely have many more of these to fix. 
```suggestion
                    out << nl << "if (";
```

Reply:
I fixed every `if(` and `switch(` I could find in the generated code.
We don't generate any `while` or `try` statements.